src/resultsbottomuppage.cpp

```diff
@@ -95,6 +95,33 @@ ResultsBottomUpPage::ResultsBottomUpPage(FilterAndZoomStack* filterStack, PerfPa
                 }
             }
         });
+
+    connect(parser, &PerfParser::bottomUpDataPerThreadAvailable, this,
+            [this](const Data::BottomUpResults& data) { bottomUpByThread = data; });
+
+    connect(parser, &PerfParser::bottomUpDataPerProcessAvailable, this,
+            [this](const Data::BottomUpResults& data) { bottomUpByProcess = data; });
+
+    connect(parser, &PerfParser::bottomUpDataPerCpuAvailable, this,
+            [this](const Data::BottomUpResults& data) { bottomUpByCpu = data; });
+
+    for (const auto& aggregationType :
+         {QLatin1String("Symbol"), QLatin1String("Thread"), QLatin1String("Process"), QLatin1String("CPU")}) {
+        ui->costAggregationComboBox->addItem(aggregationType);
+    }
+
+    connect(ui->costAggregationComboBox, QOverload<int>::of(&QComboBox::currentIndexChanged), this, [this] {
+        const auto currentText = ui->costAggregationComboBox->currentText();
+        if (currentText == QLatin1String("Symbol")) {
```

Comment:
introduce an enum and set that as data and extract it here via currentData, then switch over it instead of doing string comparisons

note e.g. that the strings should be translatable which would then break here